lib/web_ui/dev/integration_tests_manager.dart

```diff
@@ -0,0 +1,345 @@
+        testResults = testResults && await _runTestsInTheDirectory(e);
+      }
+    }
+    return testResults;
+  }
+
+  /// Run tests in a single directory under: e2e_tests/web.
+  ///
+  /// Run `flutter pub get` as the first step.
+  ///
+  /// Validate the directory before running the tests. Each directory is
+  /// expected to be a test project which includes a `pubspec.yaml` file
+  /// and a `test_driver` directory.
+  Future<bool> _runTestsInTheDirectory(io.Directory directory) async {
+    final bool directoryContentValid = _validateTestDirectory(directory);
+    if (directoryContentValid) {
+      await _runPubGet(directory.path);
+      final bool testResults = await _runTestsInDirectory(directory);
+      return testResults;
+    } else {
+      return false;
+    }
+  }
+
+  Future<bool> _runTestsInDirectory(io.Directory directory) async {
+    final io.Directory testDirectory =
+        io.Directory(pathlib.join(directory.path, 'test_driver'));
+    final List<io.FileSystemEntity> entities =
+        testDirectory.listSync(followLinks: false);
+
+    final List<String> e2eTestsToRun = List<String>();
+
+    // The following loops over the contents of the directory and saves an
+    // expected driver file name for each e2e test assuming any dart file
+    // not ending with `_test.dart` is an e2e test.
+    // Other files are not considered since developers can add files such as
+    // README.
+    for (io.File f in entities) {
+      final String basename = pathlib.basename(f.path);
+      if (!basename.contains('_test.dart') && basename.endsWith('.dart')) {
+        e2eTestsToRun.add(basename);
+      }
+    }
+
+    print(
+        'INFO: In project ${directory} ${e2eTestsToRun.length} tests to run.');
+
+    int numberOfPassedTests = 0;
+    int numberOfFailedTests = 0;
+    for (String fileName in e2eTestsToRun) {
+      final bool testResults = await _runTestsInDebugMode(directory, fileName);
+      if (testResults) {
+        numberOfPassedTests++;
+      } else {
+        numberOfFailedTests++;
+      }
+    }
+    final int numberOfTestsRun = numberOfPassedTests + numberOfFailedTests;
+
+    print('${numberOfTestsRun} tests run ${numberOfPassedTests} passed and '
+        '${numberOfFailedTests} failed.');
+    return numberOfFailedTests == 0;
+  }
+
+  Future<bool> _runTestsInDebugMode(
+      io.Directory directory, String testName) async {
+    final String statementToRun = 'flutter drive '
+        '--target=test_driver/${testName} -d web-server --release '
+        '--browser-name=$_browser --local-engine=host_debug_unopt';
+    print('INFO: To manually run the test use $statementToRun under '
+          'directory ${directory.path}');
+    // TODO(nurhan): Give options to the developer to run tests in another mode.
+    final int exitCode = await runProcess(
+      'flutter',
+      <String>[
+        'drive',
+        '--target=test_driver/${testName}',
+        '-d',
+        'web-server',
+         '--profile',
```

Comment:
ah sorry! I changed my mind on the way, forgot to change the code. 
Here is the story: release is very useful but it removes the stack trace from the errors. I wanted to use debug, but then I realized temporarily debug mode is not available. @jonahwilliams can give more detail. I think it will be available in the next releases.

I'll make it profile mode. I will also add a flag for release mode. One can debug using flutter run currently.

Reply:
Yeah, I think profile mode is a good default choice. These tests will primarily test correctness of release mode code, and debug mode uses an entirely separate compiler (so we can't trust the outcome). Profile mode uses the same compiler as release mode. It just keeps symbols unobfuscated.

Reply:
thanks!